### ISSUE TITLE
Deploy fails using http.alias

### ISSUE THREAD

**Issue by josephphayden** [aws/copilot-cli#2503]
I've got my app deployed up and running on my custom domain, having used the --domain flag on init. It was available on:
https://api.staging.myapp.mydomain.uk

When I try adding an alias to get it on `staging-api.mydomain.uk`, I get the following error:

HTTPListenerRuleWithDomain
UPDATE_FAILED 
CustomResource attribute error: Vendor response doesn't contain Priority key in object arn:aws:cloudformation:eu-west-2:xxxxxxxxxxx:stack/myapp-staging-api/085836e0-d377-11eb-9085-0adbd0eea14c\|HTTPSRulePriorityAction\|3e1562d3-192c-4070-a54a-915156cb4d4e in S3 bucket cloudformation-custom-resource-storage-euwest2

Any help much appreciated!

### copilot version
1.8.0

### manifest.yml
~~~
# The manifest for the "api" service.
# Read the full specification for the "Load Balanced Web Service" type at:
#  https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/

# Your service name will be used in naming your resources like log groups, ECS services, etc.
name: api
type: Load Balanced Web Service

# Distribute traffic to your service.
http:
  # Requests to this path will be forwarded to your service.
  # To match all requests you can use the "/" path.
  path: '/'
  # You can specify a custom health check path. The default is "/".
  # healthcheck: '/'

# Configuration for your containers and service.
image:
  # Docker build arguments. For additional overrides: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#image-build
  build: api/Dockerfile
  # Port exposed through your container to route traffic to it.
  port: 8000

cpu: 256 # Number of CPU units for the task.
memory: 512 # Amount of memory in MiB used by the task.
count: 1 # Number of tasks that should be running in your service.
exec: true # Enable running commands in your container.

# Optional fields for more advanced use-cases.
#
#variables:                    # Pass environment variables as key value pairs.
#  LOG_LEVEL: info

#secrets:                      # Pass secrets from AWS Systems Manager (SSM) Parameter Store.
#  GITHUB_TOKEN: GITHUB_TOKEN  # The key is the name of the environment variable, the value is the name of the SSM parameter.

# You can override any of the values defined above by environment.
environments:
  staging:
    http:
      alias: staging-api.mydomain.uk
    secrets:
      GOOGLE_AUTH_CLIENT_ID: xxx
      GOOGLE_AUTH_CLIENT_SECRET: xxx
      LINKEDIN_AUTH_CLIENT_ID: xxx
      LINKEDIN_AUTH_CLIENT_SECRET: xxx
~~~

**Comment by josephphayden** [comment]
It seems to work after I delete my service and recreate it with the alias in there, whereas originally I had tried to add the alias to the existing service.

**Comment by iamhopaul123** [comment]
Hello @josephphayden. It seems like most likely `HTTPSRulePriorityAction` was updated and for an update action `HTTPSRulePriorityAction` won't do anything, causing  `Priority` not included in the output of the lambda. Normally we won't update `HTTPSRulePriorityAction` once we created it unless `HTTPSListener` in the env stack was substituted by a new one. Do you remember if you customized your load balancer before the last deployment? Thank you!

**Comment by AkifumiSato** [comment]
I have the same problem.
When deploying with alias specified first, it seems that the health check always fails with http status: 503.
Looking at the task log, it seems that it is responding with http status: 200., is it a problem on the ALB side?

## command
```
copilot app init --domain example.com
copilot env init --name production
copilot svc init --name api
copilot svc deploy --name api
```

## manifest.yml
```yml
# The manifest for the "api" service.
# Read the full specification for the "Load Balanced Web Service" type at:
#  https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/

# Your service name will be used in naming your resources like log groups, ECS services, etc.
name: api
type: Load Balanced Web Service

# Distribute traffic to your service.
http:
  # Requests to this path will be forwarded to your service.
  # To match all requests you can use the "/" path.
  path: '/'
  alias: api.example.com
  healthcheck:
    path: '/'
    success_codes: '200,304'
    interval: 10s

# Configuration for your containers and service.
image:
  # Docker build arguments. For additional overrides: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#image-build
  build: Dockerfile
  # Port exposed through your container to route traffic to it.
  port: 4200

cpu: 256 # Number of CPU units for the task.
memory: 512 # Amount of memory in MiB used by the task.
count: 1 # Number of tasks that should be running in your service.
exec: true # Enable running commands in your container.

# Optional fields for more advanced use-cases.
variables:
  PORT: 4200
#secrets:                      # Pass secrets from AWS Systems Manager (SSM) Parameter Store.
#  GITHUB_TOKEN: GITHUB_TOKEN  # The key is the name of the environment variable, the value is the name of the SSM parameter.

# You can override any of the values defined above by environment.
#environments:
#  test:
#    count: 2               # Number of tasks to run for the "test" environment.
```

**Comment by josephphayden** [comment]
Thanks @iamhopaul123, I hadn't customised anything.

**Comment by iamhopaul123** [comment]
Hello @AkifumiSato did you have the same issue with very similar error message as
```
CustomResource attribute error: Vendor response doesn't contain Priority key in object arn:aws:cloudformation:eu-west-2:xxxxxxxxxxx:stack/myapp-staging-api/085836e0-d377-11eb-9085-0adbd0eea14c|HTTPSRulePriorityAction|3e1562d3-192c-4070-a54a-915156cb4d4e in S3 bucket cloudformation-custom-resource-storage-euwest2
```
or in what step did it fail to deploy?

**Comment by AkifumiSato** [comment]
@iamhopaul123 I'm sorry, I didn't leave an error log.
However, during execution, there is a log on the task side that processed the request with 200, but I saw that the task was restarted many times due to `unhealthy` health check.

**Comment by iamhopaul123** [comment]
Hello @josephphayden and @AkifumiSato. So sorry for the churns. We just released [v1.8.2](https://github.com/aws/copilot-cli/releases/tag/v1.8.2) with a couple of fixes for our lambda custom resources. Hopefully that could help. Feel free to reopen this issue or create a new one if you believe it is another issue!
